src/gui/updater/updatedownloadeddialog.cpp

```diff
@@ -44,9 +44,14 @@ UpdateDownloadedDialog::UpdateDownloadedDialog(QWidget *parent, const QString &s
     yesButton->setDefault(true);
 }
 
-UpdateDownloadedDialog::~UpdateDownloadedDialog()
+void UpdateDownloadedDialog::accept()
 {
-    delete _ui;
+    Q_EMIT accepted();
+    close();
```

Comment:
We should rather delete the class in the caller than rely on delete on close.

Reply:
Do memory management on the creating side, but check the signals.